src/database/volatile/mod.rs

```diff
@@ -16,33 +17,44 @@ use crate::{
     model::State,
 };
 
-pub struct Database<'a> {
-    memory: HashMap<State, &'a [u8]>,
+/// The Database struct holds an in-memory hashmap from States to BitVec values (derived from Records)
```

Comment:
Thank you so much for writing gramatically correct and helpful docstrings. In the spirit of improvement, try sticking to the right level of abstraction. For example, you could say:

```
/// [`KVStore`] implementation backed by a [`HashMap`]. Performance-relevant
/// information about how you handle tables, asymptotics, etc. Space constraints.
/// Considerations (e.g., "if the size exceeds X then Y will happen, impacting
/// performance by Z").
```

...instead of including private information, like how values are stored under the hood.